src/app/icd/BUILD.gn

```diff
@@ -23,6 +23,7 @@ source_set("manager-srcs") {
   sources = [
     "ICDManager.cpp",
     "ICDManager.h",
+    "${chip_root}/src/app/reporting/ReportScheduler.h",
```

Comment:
Isn't that strange to "build" this header here? Shouldn't it be a deps?

Reply:
This was trying to work around the fact that the ReportScheduler is in the sources of src/app/BUILD.gn, which depends on the icd manager. So to get this source we would need to add src/app/ in the dependencies but that would create a circular dependency so the source was added here directly.
Since the ReportScheduler requires the ReadHandler, it is not possible to place it in an external dependency so another workaround will be found.